modules/animation/include/modules/animation/datastructures/propertytrack.h

```diff
@@ -361,7 +413,11 @@ inline std::unique_ptr<Key> PropertyTrack<Prop, Key, Seq>::createKeyframe(Second
 template <typename Prop, typename Key, typename Seq>
 inline std::unique_ptr<Key> PropertyTrack<Prop, Key, Seq>::createKeyframe(const Prop* property,
                                                                           Seconds time) const {
-    return std::make_unique<Key>(time, property->get());
+    if constexpr (detail::property_has_get<Prop, Key::value_type>::value) {
+        return std::make_unique<Key>(time, property->get());
+    } else {
+        return std::make_unique<Key>(time);
```

Comment:
is this case not an error?

Reply:
No, not all properties have get functions, e.g. ButtonProperty.

Reply:
Ahh I see...